pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/TransactionBufferProvider.java

```diff
@@ -51,5 +52,5 @@ static TransactionBufferProvider newProvider(String providerClassName) throws IO
      * @param originTopic
      * @return
      */
-    TransactionBuffer newTransactionBuffer(Topic originTopic);
+    TransactionBuffer newTransactionBuffer(Topic originTopic, PositionImpl startUsedPosition);
```

Comment:
Maybe we do not need to add the new API. We can require the startUsedPosition in the Transaction Buffer again.
It is still a break-changing even though we added a new default method because we changed the logic in the PersistentTopic that called this method.